src/main/java/mekanism/client/gui/GuiMekanismTile.java

```diff
@@ -39,7 +41,8 @@ protected void addGenericTabs() {
             upgradeWindowTab = addRenderableWidget(new GuiUpgradeWindowTab(this, tile, () -> upgradeWindowTab));
         }
         if (tile.supportsRedstone()) {
-            addRenderableWidget(new GuiRedstoneControlTab(this, tile));
+            BooleanSupplier supplier = trackWarning(WarningType.REDSTONE_UNACTIVATED, () -> !tile.isRedstoneActivated());
+            addRenderableWidget(new GuiRedstoneControlTab(this, tile).warning(supplier));
```

Comment:
This should probably be adjusted to just be:
```java
addRenderableWidget(new GuiRedstoneControlTab(this, tile).warning(WarningType.REDSTONE_UNACTIVATED, () -> !tile.isRedstoneActivated()));
```
and have the `warning` method be implemented similar to the one in `GuiSlot`